File: src/1877.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int maxPair = 0;
        int i = 0, j = nums.size() - 1;
        while (i < j) {
            maxPair = max(nums[i] + nums[j], maxPair);
            i++;
            j--;
        }
        return maxPair;
    }
};

File: src/1877.py
class Solution:
    def minPairSum(self, nums: list[int]) -> int:
        nums.sort()
        maxPair = 0
        i = 0 
        j = len(nums) - 1
        while i < j:
            maxPair = max(nums[i] + nums[j], maxPair)
            i += 1
            j -= 1
        return maxPair
